core/src/main/java/feign/Response.java

```diff
@@ -327,6 +332,11 @@ public Reader asReader() throws IOException {
       return new InputStreamReader(inputStream, UTF_8);
     }
 
+    @Override
+    public Reader asReader(Charset charset) throws IOException {
+      return new InputStreamReader(inputStream, charset == null ? UTF_8 : charset);
```

Comment:
Let's make `charset` mandatory when calling this method.

Use `Util.checkNotNull(charset, "'charset' must be not null")`

Reply:
I will create a new pr